### PR TITLE
Remove tagstore.

### PR DESCRIPTION
**Pull request by symbolist**
## Description

Tagstore is not being used so the code is being removed. Systems using blockstore for content storage should develop their own tagging system appropriate for their needs.

## Test Instructions

The tagstore code is not being used anywhere. Running the end-to-end tests is sufficient to confirm that the bundles code is working as expected.

To test the migrations:
```
git checkout master
docker exec -t edx.devstack.mysql57 /bin/bash -c 'mysql -uroot <<< "drop database blockstore_db;"'
docker exec -t edx.devstack.mysql57 /bin/bash -c 'mysql -uroot <<< "create database if not exists blockstore_db;"'
make blockstore-shell
./manage migrate
./manage dbshell
mysql> show tables; # This will print all tables including `tagstore_entity`, `tagstore_entity_tags`, `tagstore_tag`, `tagstore_taxonomy`.
exit
exit

git checkout symbolist/remove-tagstore
make blockstore-shell
./manage.py migrate # This will apply tagstore_django.0002_delete_all_app_models.
./manage dbshell
mysql> show tables; # The tagstore_ tables will no longer show in the list.
```

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @symbolist! I've created [OSPR-5127](https://openedx.atlassian.net/browse/OSPR-5127) to keep track of it in JIRA.

As a core committer in this repo, you can merge this once the pull request is approved per the [core committer reviewer requirements](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529675973/Open+edX+Core+Committers#Reviews-and-Ownership) and according to the agreement with your edX Champion.

File: decisions/0003-tagstore-removal.rst
Comment (bradenmacdonald) [inline decisions/0003-tagstore-removal.rst:13]: The reason was more specific than that - it was to store the Taxonomies as versioned data in bundles:

> Currently [in Tagstore], taxonomies of tags are stored in MySQL and use "materialized paths" to represent hierarchy where present. Presumably, when XBlock/course content in Blockstore is moved, duplicated, deleted, etc., Tagstore will listen for signals from Blockstore and then update its database accordingly. One problem I have yet to solve is how to include tags in the bundle when exporting courses etc. for import into other Open edX instances (we want to reference the taxonomy that the tags come from, and import it as well but not if it's already present). I also realize that the current implementation makes it difficult for draft/publish support with tags - tags are always applied immediately since Tagstore isn't aware of draft content vs. published content.
>
> I realized today that a potential solution to most of these issues is **to store the tag data in files in a Bundle instead of in MySQL**. I had initially rejected that option because it doesn't support hierarchical querying of the tag data (searching for "animal" should return things tagged "lion"), but I've since realized that we'll likely want to index the tag data in ElasticSearch anyways, and as long as we index the materialized paths in ElasticSearch, it doesn't matter how they're stored in the original tag/taxonomy data source.
>
> ...
> So we're talking about a dedicated Bundle type that is a Taxonomy, that's edited, published, and linked to by other Bundles? I like it.

**Comment by bradenmacdonald** [comment]
Thanks for this, sorry I didn't get to it myself! I want to suggest one thing though: currently many devstacks and even production instances have the tagstore tables in MySQL. We should probably create, publish, and deploy a migration that deletes those tables before we remove the tagstore migrations/models completely. Thoughts?

**Comment by eLRuLL** [comment]
I agree with @bradenmacdonald that should be pretty helpful. Apart from that the change looks very straightforward, :+1: 

- [ ] I read through the code and found no concers
- [ ] I didn't test this locally, but made sure LX doesn't use tagstore
- [ ] The code includes documentation/comments

**Comment by symbolist** [comment]
Thanks @eLRuLL!

@bradenmacdonald 

> Edit: on second thought, if we're moving blockstore into Studio and deleting the entire blockstore database eventually, it really doesn't matter. So maybe just go ahead with this :)

Yup, that was my assumption too.

@ormsbee @kdmccormick Want to confirm if the assumption that we will switch to using the edxapp database for blockstore once it is converted to an app correct?

**Comment by kdmccormick** [comment]
@symbolist Great question. I also assume we'd migrate the remaining Blockstore tables to the edxapp database, unless there was some benefit to keeping the them in their own db (and then pointing Studio at the Blockstore database).

@ormsbee Can you think of any reason we wouldn't migrate the Blockstore tables into the `edxapp` database when we install it into Studio as an app?

**Comment by natabene** [comment]
@bradenmacdonald I assume you are doing fine without me here.

**Comment by natabene** [comment]
@bradenmacdonald @kdmccormick Just curious if there is any way to push this forward, or it this PR still blocked?

**Comment by bradenmacdonald** [comment]
I'm not sure - @symbolist what's the status of this? I guess if we're moving blockstore into edx-platform soon we don't need this?

**Comment by symbolist** [comment]
@natabene @bradenmacdonald This is a cleanup in preparation for https://github.com/edx/blockstore/pull/97 I have been caught up the last few weeks but planning to get all this ready in the coming weeks. For now I have marked this a draft.

File: decisions/0003-tagstore-removal.rst
Comment (symbolist) [reply decisions/0003-tagstore-removal.rst]: @bradenmacdonald Updated. Does this work?

Comment (bradenmacdonald) [reply decisions/0003-tagstore-removal.rst]: @symbolist Looks good, thanks!

**Comment by symbolist** [comment]
@bradenmacdonald I updated this so it deletes the tagstore tables from the db. Can you do a final review? Thanks!

**Review by bradenmacdonald — APPROVED** [review]
:+1:

- [x] I tested this: as described, it deleted the tagstore tables.
- [x] I read through the code
- [x] I checked for accessibility issues: n/a
- [x] Includes documentation

**Comment by edx-webhook** [comment]
@symbolist Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/blockstore&entry.867055334=https://github.com/edx/blockstore/pull/96&entry.1484655318=https://github.com/symbolist&entry.752974735=2020-11-11+22:56&entry.1917517419=2021-09-30+12:09&entry.2133058324=No) so we can improve your experience in the future.

**Comment by symbolist** [comment]
@kdmccormick Hmm, this also seems to be stuck and I see even new PRs are seeing the CircleCI jobs e.g. https://github.com/edx/blockstore/pull/129

Maybe the "Stop Building" option on this page is what is needed?

https://app.circleci.com/settings/project/github/edx/blockstore?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fedx%2Fblockstore

**Comment by kdmccormick** [comment]
@symbolist Turns out the branch protection rules needed changing -- the CircleCI jobs needed to be removed from the list of required checks, and the GHA checks needed to be added to that list. Should be all good now.

**Comment by edx-webhook** [comment]
@ormsbee, @kdmccormick: thought you might like to know that symbolist merged this pull request.

**Comment by symbolist** [comment]
@kdmccormick Thanks! Can we get master branch deployed to production? I am working on the [Django upgrade](https://github.com/edx/blockstore/pull/128) so we can deploy that independently.

**Comment by kdmccormick** [comment]
@symbolist On it!

**Pull request merged**
